aten/src/ATen/native/cudnn/LossCTC.cpp

```diff
@@ -211,19 +247,92 @@ std::tuple<Tensor, Tensor> _cudnn_ctc_loss(
 }
 
 std::tuple<Tensor, Tensor> _cudnn_ctc_loss_tensor(
-    const Tensor& log_probs,
-    const Tensor& targets,
+    const Tensor& log_probs_t,
+    const Tensor& targets_t,
     const Tensor& input_lengths,
     const Tensor& target_lengths,
     int64_t BLANK,
     bool deterministic,
     bool zero_infinity) {
-  Tensor ilc = input_lengths.to(Device(at::kCPU), at::kLong).contiguous();
-  Tensor tlc = target_lengths.to(Device(at::kCPU), at::kLong).contiguous();
-  IntArrayRef il(ilc.data_ptr<int64_t>(), ilc.numel());
-  IntArrayRef tl(tlc.data_ptr<int64_t>(), tlc.numel());
-  return at::_cudnn_ctc_loss(
-      log_probs, targets, il, tl, BLANK, deterministic, zero_infinity);
+  Tensor targets_t_ = targets_t;
+  if (targets_t.device().type() == at::kCPU) {
+    targets_t_ = targets_t.to(Device(at::kCUDA));
+  }
+  const CheckedFrom c = "cudnn_ctc_loss";
+  const TensorArg log_probs{log_probs_t, "log_probs", 1};
+  const TensorArg targets{targets_t_, "targets", 2};
+  checkDim(c, log_probs, 3);
+  checkScalarType(c, log_probs, kFloat);
+  checkDim(c, targets, 1);
+  checkScalarType(c, targets, kInt);
+  checkContiguous(c, targets); // ?
+  checkBackend(c, {*log_probs}, Backend::CUDA);
+  checkBackend(c, {*targets}, Backend::CUDA);
+  const auto batch_size = log_probs->size(1);
+  int64_t input_lengths_size =
+      input_lengths.sizes().size() ? input_lengths.size(0) : 1;
+  int64_t target_lengths_size =
+      target_lengths.sizes().size() ? target_lengths.size(0) : 1;
+  TORCH_CHECK(
+      input_lengths_size == batch_size,
+      "input_lengths needs to have size to match batch_size");
+  TORCH_CHECK(
+      target_lengths_size == batch_size,
+      "target_lengths needs to have size to match batch_size");
+
+  TORCH_CHECK(BLANK == 0, "blank must be label 0 for cudnn_ctc_loss");
+  // checked in dispatch:
+  // assert other conditions for cudnnCTCLoss: all label lengths <= 256
+  // all input lengths = logprob.size(0)
+
+  const auto handle = getCudnnHandle();
+
+  const cudnnCTCLossAlgo_t algo =
+      (deterministic ? CUDNN_CTC_LOSS_ALGO_DETERMINISTIC
+                     : CUDNN_CTC_LOSS_ALGO_NON_DETERMINISTIC);
+
+  CTCLossDescriptor ctc_loss_desc;
+
+  // so the CuDNN gradient semantics have changed between 7.1 and 7.6,
+  // this is CuDNN 7.6 only, see PyTorch 1.2 for older CuDNN.
```

Comment:
Isn't there a typo of sorts? I.e. it calls `set_v9` function online 298, but at the same type test says that it is supported on v7.6+